src/odemis/acq/align/spot.py

```diff
@@ -357,31 +358,46 @@ def FindGridSpots(image, repetition):
     Returns
     -------
     spot_coordinates : array like
-        A 2D array of shape (N, 2) containing the coordinates of the spots.
+        A 2D array of shape (N, 2) containing the coordinates of the spots,
+        in respect to the top left of the image.
     translation : tuple of two floats
+        Translation from the origin to the center of the grid in image space,
+        origin is top left of the image. Primary axis points down and the
+        secondary axis points to the right.
     scaling : tuple of two floats
+        Scaling factors for primary and secondary axis.
     rotation : float
+        Rotation in image space, positive rotation is clockwise.
     shear : float
+        Horizontal shear factor. A positive shear factor transforms a coordinate
+        in the positive x direction parallel to the x axis.
 
     """
+    # Find the center coordinates of the spots in the image.
     spot_positions = MaximaFind(image, repetition[0] * repetition[1])
     if len(spot_positions) < repetition[0] * repetition[1]:
         logging.warning('Not enough spots found, returning only the found spots.')
         return spot_positions, None, None, None
-    # Estimate transformation
+    # Estimate the two most common directions in the grid of spots, these form the transformation matrix.
     lattice_constants = EstimateLatticeConstant(spot_positions)
+    # Transpose to put the primary axis on the first row instead of the first column. In a transformation matrix, the
+    # first row should represent the transform for the primary axis, second row for the secondary, etc..
     transformation_matrix = numpy.transpose(lattice_constants)
     if numpy.linalg.det(lattice_constants) < 0.:
+        # If the determinant is smaller than zero flip the transformation matrix to make it a right handed
+        # transformation matrix.
```

Comment:
You don't really need to say "If the determinant is smaller than zero" because that's what the code just above says. Either don't mention it, or "translate" it into something more high-level (eg, "A negative determinant means that the first axis goes blabla compared to the second axis"). I'm not convinced that saying "right handed transformation matrix" is really clear. Maybe you can say that "we assume that the image is not inverted and not too rotated, so the horizontal axis should be the first axis (X) and the vertical axis the second axis (Y). I'm not sure that's true though... I leave it up to you ;-)

Reply:
I will think of something better! 

I also wonder if it makes sense to move this if statement inside `EstimateLatticeConstant`, then `fliplr` should be changed to `flipud`. Moving it inside `EstimateLatticeConstant` will make the output consistent. Now the output of `EstimateLatticeConstant` depends on the "strongest" mean, found by the kmeans algorithm.

Reply:
@Sabrina85 what do you think of moving `fliplr` inside `EstimateLatticeConstant`? That way the output of `EstimateLatticeConstant` has the horizontal axis first and vertical axis second, corresponding to the axes of `spot_positions`; the input of `EstimateLatticeConstant`.

Reply:
That makes sense! What about the `transformation_matrix = numpy.transpose(lattice_constants)`? That would need to be moved as well as it does the transform as described in the comment?